src/reader.rs

```diff
@@ -1,79 +1,57 @@
 use std::io;
 use std::ops::Range;
 
-use bigdecimal::BigDecimal;
-use chrono::{DateTime, FixedOffset};
 use delegate::delegate;
 
 use crate::constants::v1_0::system_symbol_ids;
-use crate::raw_reader::RawStreamItem::*;
+use crate::raw_reader::{RawReader, RawStreamItem};
 use crate::raw_symbol_token::RawSymbolToken;
-use crate::result::{decoding_error, IonResult};
-use crate::symbol_table::SymbolTable;
+use crate::result::{decoding_error, decoding_error_raw, IonResult};
+use crate::stream_reader::StreamReader;
+use crate::symbol_table::{Symbol, SymbolTable};
 use crate::types::decimal::Decimal;
 use crate::types::timestamp::Timestamp;
-use crate::{IonType, RawBinaryReader, RawReader};
+use crate::{IonType, RawBinaryReader};
 
-/// A streaming Ion reader that resolves symbol IDs into the appropriate text.
+/// A streaming Ion reader that resolves symbol IDs into their corresponding text.
 ///
 /// Reader itself is format-agnostic; all format-specific logic is handled by the
-/// wrapped Cursor implementation.
+/// wrapped [RawReader] implementation.
 pub struct Reader<R: RawReader> {
     raw_reader: R,
     symbol_table: SymbolTable,
 }
 
-// FIXME: The `read_datetime` method is deprecated. However, this function body is generated by a
```

Comment:
PR tour: the application-level `Reader` has been changed from a standalone struct to a struct that implements `StreamReader`.